codex/blockexchange/engine/advertiser.nim

```diff
@@ -124,6 +124,10 @@ proc start*(b: Advertiser) {.async: (raises: []).} =
 
   trace "Advertiser start"
 
+  if b.advertiserRunning:
+    warn "Starting advertiser twice"
```

Comment:
I think there is never a situation in which the advertiser should be started twice. If that's correct, I propose we put a raiseAssert here. This way the node will crash and we'll have to fix it.

Reply:
I think that's a good plan. I would like to see more raiseAsserts in general for situations that should not happen. I think this is a prime example.